--- a/driver/plugins.cpp	2026-02-06 18:17:26.847764358 +0100
+++ b/driver/plugins.cpp	2026-02-06 18:18:17.759283880 +0100
@@ -25,7 +25,11 @@
 #include "llvm/Support/CommandLine.h"
 #include "llvm/Support/DynamicLibrary.h"
 #include "llvm/ADT/SmallVector.h"
+#if LDC_LLVM_VER >= 2100
+#include "llvm/Plugins/PassPlugin.h"
+#else
 #include "llvm/Passes/PassPlugin.h"
+#endif
 #include "llvm/Support/Error.h"
 
 #include "driver/cl_options.h"
--- a/driver/cl_options.cpp	2026-02-06 18:13:47.410004303 +0100
+++ b/driver/cl_options.cpp	2026-02-06 18:20:20.678105478 +0100
@@ -788,12 +788,20 @@
 /// The clashing LLVM options are suffixed with "llvm-" and hidden from the
 /// -help output.
 void createClashingOptions() {
+#if LDC_LLVM_VER >= 2200
+  llvm::DenseMap<llvm::StringRef, cl::Option *> &map = cl::getRegisteredOptions();
+#else
   llvm::StringMap<cl::Option *> &map = cl::getRegisteredOptions();
+#endif
 
   auto renameAndHide = [&map](const char *from, const char *to) {
     auto i = map.find(from);
     if (i != map.end()) {
+#if LDC_LLVM_VER >= 2200
+      cl::Option *opt = i->getSecond();
+#else
       cl::Option *opt = i->getValue();
+#endif
       map.erase(i);
       if (to) {
         opt->setArgStr(to);
@@ -1008,7 +1016,11 @@
                                                "trap-func",
                                                "W"};
 
+#if LDC_LLVM_VER >= 2200
+  llvm::DenseMap<llvm::StringRef, cl::Option *> &map = cl::getRegisteredOptions();
+#else
   llvm::StringMap<cl::Option *> &map = cl::getRegisteredOptions();
+#endif
   for (const auto name : hiddenOptions) {
     // Check if option exists first for resilience against LLVM changes
     // between versions.
